lua/zettelkasten/browser.lua

```diff
@@ -1,42 +1,65 @@
 local M = {}
 local fn = vim.fn
 
+local log_levels = vim.log.levels
+local log = require("zettelkasten.log")
 local config = require("zettelkasten.config")
 
-local ZK_FULL_TITLE_PATTERN = "# %d+-%d+-%d+-%d+-%d+-%d+ .+"
-local ZK_ID_PATTERN = "%d+-%d+-%d+-%d+-%d+-%d+"
-local ZK_FILE_NAME_PATTERN = "%d+-%d+-%d+-%d+-%d+-%d+.md"
-
 local s_note_cache_with_file_path = {}
 local s_note_cache_with_id = {}
 
-local function get_files(folder)
+local function get_files(folder, cfg)
     local files = fn.split(fn.globpath(folder, "*.md"), "\\n")
     files = vim.tbl_filter(function(file)
-        return string.match(file, ZK_FILE_NAME_PATTERN) ~= nil
+        return string.match(file, cfg.filename_pattern) ~= nil
     end, files)
 
     return files
 end
 
-local function extract_id_and_title(line)
-    local zk_id = string.match(line, ZK_FULL_TITLE_PATTERN)
+local function extract_id_and_title(line, file_path, cfg)
+    local zk_id = string.match(line, cfg.title_pattern)
     if zk_id == nil then
         return nil
     end
 
+    local file_id = nil
+    if cfg.id_in_filename then
+        file_id = string.match(vim.fn.fnamemodify(file_path,':t:r'), cfg.id_pattern)
+    end
+
     zk_id = string.gsub(zk_id, "# ", "")
-    local note_id = string.match(zk_id, ZK_ID_PATTERN)
-    local title = vim.trim(string.gsub(zk_id, ZK_ID_PATTERN, ""))
+    local title_id = nil
+    local title = nil
+    if cfg.id_in_title then
+        title_id = string.match(zk_id, cfg.id_pattern)
+        title = vim.trim(string.gsub(zk_id, cfg.id_pattern, ""))
+    else
+        title = vim.trim(zk_id)
+    end
+
+    local note_id = nil
+    if cfg.id_in_filename and cfg.id_in_title then
+        if file_id ~= title_id then
+            log.notify('Filename id "'..file_id..'" and title id "'..title_id..'" did not match.', log_levels.ERROR, {})
```

Comment:
This prevents the browser from being loaded. I'm slightly worried about the effect this might have when there's multiple files with mismatching `id` in the title and the file name. Maybe you can append something like `[ID-MISMATCH]` to the title of the note so that the browser loads all the notes and one can easily see all the notes with the mismatch and correct them. Otherwise when there's more than one, it'd be frustrating to fix.